src/miral/display_configuration.cpp

```diff
@@ -120,6 +121,117 @@ void miral::DisplayConfiguration::add_output_attribute(std::string const& key)
     self->add_output_attribute(key);
 }
 
+auto miral::DisplayConfiguration::layout_userdata(std::string const& key) const
+    -> std::optional<std::any const>
+{
+    return self->layout_userdata(key);
+}
+
+struct miral::DisplayConfiguration::Node::Self
+{
+    explicit Self(YAML::Node const& node)
+        : node(node)
+    {
+    }
+
+    ~Self() = default;
+
+    template <typename T>
+    auto as() const -> T
+    {
+        try
+        {
+            return node.as<T>();
+        }
+        catch (YAML::Exception const&)
+        {
+            return T();
+        }
+    }
+
+    YAML::Node const node;
+};
+
+miral::DisplayConfiguration::Node::Node(std::unique_ptr<Self> self)
+    : self(std::move(self))
+{
+}
+
+miral::DisplayConfiguration::Node::~Node() = default;
+
+auto miral::DisplayConfiguration::Node::type() const -> Type
+{
+    if (self->node.IsScalar())
+    {
+        try {
+            (void)self->node.as<int>();
+            return Type::integer;
+        } catch (const YAML::BadConversion& e) {
+            return Type::string;
+        }
+    }
+    else if (self->node.IsMap())
+        return Type::map;
+    else if (self->node.IsSequence())
+        return Type::sequence;
+    else
+    {
+        mir::log_error("Checked type is none of the supported types");
```

Comment:
I don't think this is an error here (we're bubbling the failure up to the caller). And we should provide as much context as possible to help a user fix the problem. (The filename and line number would be ideal, but I don't think that's available)

```suggestion
        mir::log_warning("Parsing custom display configuration node: Checked type is none of the supported types");
```